src/test/java/build/buildfarm/server/BuildFarmServerTest.java

```diff
@@ -332,4 +333,25 @@ private Operation executeAction(Action action) {
 
     return executeStub.execute(executeRequest);
   }
+
+  @Test
+  public void actionNotCached() {
+    DigestUtil digestUtil = new DigestUtil(DigestUtil.HashFunction.SHA256);
+    GetActionResultRequest request = GetActionResultRequest.newBuilder()
+        .setInstanceName("memory")
+        .setActionDigest(digestUtil.empty())
+        .build();
+
+    ActionCacheGrpc.ActionCacheBlockingStub actionCacheStub =
+        ActionCacheGrpc.newBlockingStub(inProcessChannel);
+
+    StatusRuntimeException notFound = null;
+    try {
+      actionCacheStub.getActionResult(request);
```

Comment:
nit: I think this is a bit more readable:
try {
  something bad
  fail("expected exception")
} catch (MyException e) {
  assertThat(e).isWhatIThought()
}